src/main/java/org/shanerx/tradeshop/commands/CommandRunner.java

```diff
@@ -533,7 +535,10 @@ public void open() {
 			sendMessage(Message.MISSING_ITEM.getPrefixed());
 			return;
 		}
-
+		
+		PlayerShopOpenEvent event = new PlayerShopOpenEvent(pSender, shop);
+		if (event.isCancelled()) return;
+		
 		boolean opened = shop.setOpen();
 		shop.saveShop();
```

Comment:
Can this be handled in the new event since we are passing it the player and shop anyways?